spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -1308,3 +1308,87 @@ for (Pageable p = Pageable.ofSize(25).sortBy(Sort.desc("yearBorn"),
   p = page.nextPageable();
 }
 ----
+
+===== Scenario: Person Entity and People Repository
+
+This keyset cursor-based pagination scenario uses the same `Person` entity and example dataset from the offset-based pagination scenario, but orders it by `name` and then by `id`,
+
+[source,json]
+----
+[
+   {"id":3, "name":"Alyse Dadson"},
+   {"id":6, "name":"Boothe Martinson"},
+   {"id":10, "name":"Boothe Martinson"},
+   {"id":2, "name":"Corri Davidou"},
+   {"id":8, "name":"Danita Pilipyak"},
+   {"id":9, "name":"Harlene Branigan"},
+   {"id":5, "name":"Jaquith Wealthall"},
+   {"id":1, "name":"Lin Le Marchant"},
+   {"id":4, "name":"Orelle Roughey"},
+   {"id":7, "name":"Patten Bedell"}
+]
+----
+
+[source,java]
+----
+@Repository
+public interface People extends BasicRepository<Person, Long> {
+    KeysetAwarePage<Person> findAll(Pageable pagination);
+}
+----
+
+Code Execution:
+
+[source,java]
+----
+@Inject
+People people;
+
+Pageable firstPageRequest = Pageable.ofSize(4).sortBy(Sort.asc("name"), Sort.asc("id"));
+KeysetAwarePage<Person> page = people.findAll(firstPageRequest);
+----
+
+Resulting Page Content:
+
+[source,json]
+----
+[
+   {"id":3, "name":"Alyse Dadson"},
+   {"id":6, "name":"Boothe Martinson"},
+   {"id":10, "name":"Boothe Martinson"},
+   {"id":2, "name":"Corri Davidou"}
+]
+----
+
+
+Deletion of an Entity:
+
+----
+// The user decides to remove one of the entities that has the same name,
+people.deleteById(10);
+----
+
+
+Next Page Execution:
+
+[source,java]
+----
+Pageable nextPageRequest = page.nextPageable();
```

Comment:
@njr-11 we are using this method; should we use cursor instead?

Reply:
> @njr-11 we are using this method; should we use cursor instead?

No, `nextPagageable` on a KeysetAwarePage/KeysetAwareSlice implicitly uses a cursor, per the JavaDoc:

```
     * <p>Returns pagination information for requesting the next page
     * in a forward direction from the current page. This method computes a
     * keyset from the last entity of the current page and includes the
     * keyset in the pagination information so that it can be used to
     * obtain the next slice in a forward direction according to the
     * sort criteria and relative to that entity.</p>
```